### PR TITLE
PR: remove rope.base.utils.pycompat

### PR DESCRIPTION
**Pull request by edreamleo**
This PR contains the minimum changes required to delete pycompat.py.

- [x] Remove all references to rope.base.utils.pycompat.
- [x] Dubious? Always define basestring as (str, bytes) in rope/refactor/patchedast.py.
- [x] Remove rope/base/utils/pycompat.py
- [x] All unit tests pass.
- [x] reindent and blacken all files.

### CODE REVIEW

**Comment by edreamleo** [comment]
@lieryan This PR contains the minimum changes required to delete pycompat. I'm not sure, however, just where (against what branch) or when you might like this PR merged. What would you like to do?

File: rope/base/prefs.py [hunk 1]
```diff
@@ -258,7 +256,9 @@ def _read(self) -> bool:
                 "__file__": config.real_path,
             }
         )
-        pycompat.execfile(config.real_path, self.run_globals)
+        with open(config.real_path) as f:
+            code = compile(f.read(),config.real_path, 'exec')
```
Comment (lieryan) [inline rope/base/prefs.py:260]:
```suggestion
            code = compile(f.read(), config.real_path, "exec")
```

**Review by lieryan — APPROVED** [review]

**Comment by lieryan** [comment]
main is good for the target for most PRs

Thanks for cleaning this file up, @edreamleo.

@all-contributors add @edreamleo for code

**Comment by edreamleo** [comment]
@lieryan I've never heard of contributions or contribution names, so I have no idea what the bot is talking about :-)

Do you have any idea what I should do?

**Comment by lieryan** [comment]
Nothing to do. I'm not sure what the bot is even complaining about, since the it created its PR correctly. 🤷

**Pull request merged**